src/Adapter/Shipment/QueryHandler/GetShipmentForViewingHandler.php

```diff
@@ -0,0 +1,103 @@
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Shipment\QueryHandler;
+
+use PrestaShop\PrestaShop\Adapter\Address\Repository\AddressRepository;
+use PrestaShop\PrestaShop\Adapter\Carrier\Repository\CarrierRepository;
+use PrestaShop\PrestaShop\Adapter\Country\Repository\CountryRepository;
+use PrestaShop\PrestaShop\Adapter\Order\Repository\OrderRepository;
+use PrestaShop\PrestaShop\Adapter\State\Repository\StateRepository;
+use PrestaShop\PrestaShop\Core\CommandBus\Attributes\AsQueryHandler;
+use PrestaShop\PrestaShop\Core\Domain\Address\QueryResult\ShippingAdressSummary;
+use PrestaShop\PrestaShop\Core\Domain\Address\ValueObject\AddressId;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\QueryResult\CarrierSummary;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\ValueObject\CarrierId;
+use PrestaShop\PrestaShop\Core\Domain\Country\ValueObject\CountryId;
+use PrestaShop\PrestaShop\Core\Domain\Order\ValueObject\OrderId;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\Exception\ShipmentNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\Query\GetShipmentForViewing;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\QueryHandler\GetShipmentForViewingHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\QueryResult\ShipmentForViewing;
+use PrestaShop\PrestaShop\Core\Domain\State\ValueObject\StateId;
+use PrestaShopBundle\Entity\Repository\ShipmentRepository;
+use PrestaShopBundle\Entity\Shipment;
+
+#[AsQueryHandler]
+class GetShipmentForViewingHandler implements GetShipmentForViewingHandlerInterface
+{
+    public function __construct(
+        private readonly ShipmentRepository $shipmentRepository,
+        private readonly OrderRepository $orderRepository,
+        private readonly CarrierRepository $carrierRepository,
+        private readonly CountryRepository $countryRepository,
+        private readonly AddressRepository $addressRepository,
+        private readonly StateRepository $stateRepository
+    ) {
+    }
+
+    /**
+     * @param GetShipmentForViewing $query
+     *
+     * @return ShipmentForViewing
+     */
+    public function handle(GetShipmentForViewing $query)
+    {
+        /** @var Shipment $shipment */
+        $shipment = $this->shipmentRepository->find($query->getShipmentId()->getValue());
+
+        if ($shipment === null) {
+            throw new ShipmentNotFoundException(sprintf('Could not find shipment with id "%s"', $query->getShipmentId()->getValue()));
+        }
+        $order = $this->orderRepository->get(new OrderId($shipment->getOrderId()));
+        $carrier = $this->carrierRepository->get(new CarrierId($shipment->getCarrierId()));
+        $address = $this->addressRepository->get(new AddressId($shipment->getAddressId()));
```

Comment:
This feels weird to create an instance from a getter with the same name. This is probably how things must be implemented on PrestaShop now, and in this case can someone let me know?